relay-event-schema/src/protocol/ourlog.rs

```diff
@@ -154,90 +159,319 @@ impl Empty for AttributeValue {
-            }
-            Annotated(Some(Value::I64(value)), meta) => {
-                Annotated(Some(AttributeValue::IntValue(value)), meta)
+            Annotated(Some(Value::Object(mut object)), meta) => {
+                let attribute_type = object
+                    .remove("type")
+                    .and_then(|v| v.value().cloned())
+                    .and_then(|v| match v {
+                        Value::String(s) => Some(s),
+                        _ => None,
+                    });
+
+                let value = object.remove("value");
+
+                match (attribute_type.as_deref(), value) {
+                    (Some("string"), Some(Annotated(Some(Value::String(string_value)), _))) => {
+                        Annotated(Some(AttributeValue::StringValue(string_value)), meta)
+                    }
+                    (Some("int"), Some(Annotated(Some(Value::String(string_value)), _))) => {
+                        let mut meta = meta;
+                        if let Ok(int_value) = string_value.parse::<i64>() {
+                            Annotated(Some(AttributeValue::IntValue(int_value)), meta)
+                        } else {
+                            meta.add_error(Error::invalid("integer could not be parsed."));
+                            meta.set_original_value(Some(Value::Object(object)));
+                            Annotated(None, meta)
+                        }
+                    }
+                    (Some("int"), Some(Annotated(Some(Value::I64(_)), _))) => {
+                        let mut meta = meta;
+                        meta.add_error(Error::expected(
+                            "64 bit integers have to be represented by a string in JSON",
+                        ));
+                        meta.set_original_value(Some(Value::Object(object)));
+                        Annotated(None, meta)
+                    }
+                    (Some("double"), Some(Annotated(Some(Value::F64(double_value)), _))) => {
+                        Annotated(Some(AttributeValue::DoubleValue(double_value)), meta)
+                    }
+                    (Some("bool"), Some(Annotated(Some(Value::Bool(bool_value)), _))) => {
+                        Annotated(Some(AttributeValue::BoolValue(bool_value)), meta)
+                    }
+                    (Some(_), Some(Annotated(Some(Value::String(unknown_value)), _))) => {
+                        Annotated(Some(AttributeValue::Unknown(unknown_value)), meta)
+                    }
+                    _ => {
+                        let mut meta = meta;
+                        meta.add_error(Error::expected(
+                            "a valid attribute value (string, int, double, bool)",
+                        ));
+                        meta.set_original_value(Some(Value::Object(object)));
+                        Annotated(None, meta)
+                    }
+                }
```

Comment:
We're now accepting:

```json
{
  "myattribute": {
     "value": "foo",
     "type": "string"
  }
}
```

Instead of: 
```
{
  "myattribute": {
     "string_value": "foo"
  }
}
```

Although we still serialize using `string_value` (for now) until the Kafka schemas are updated / replaced by EAP items.